packages/mobile/src/exchange/ExchangeReview.tsx

```diff
@@ -1,133 +1,229 @@
+        this.props.exchangeRatePair,
+        this.state.makerToken,
+        this.state.inputToken
+      )
+      getTakerAmount(input, exchangeRate)
     }
-    return confirmationInput
+    return input
   }
 
   componentDidMount() {
-    const { makerToken, makerAmount } = this.getConfirmationInput()
-    this.props.fetchExchangeRate(makerAmount, makerToken)
+    this.getExchangePropertiesFromNavProps()
   }
 
-  renderHeader = () => {
-    return <ReviewHeader title={this.props.t('reviewExchange')} />
+  getInputAmountInToken(token: CURRENCY_ENUM) {
+    let amount = this.state.inputAmount
+    if (this.state.inputToken !== token) {
+      const conversionRate = getRateForMakerToken(
+        this.props.exchangeRatePair,
+        this.state.makerToken,
+        this.state.inputToken
+      )
+      amount = getTakerAmount(amount, conversionRate)
+    }
+    return amount
   }
 
   render() {
-    const { exchangeRatePair, fee, t, appConnected, dollarBalance, goldBalance } = this.props
-    const { makerToken, makerAmount } = this.getConfirmationInput()
-    const rate = getRateForMakerToken(exchangeRatePair, makerToken)
-    const takerAmount = getTakerAmount(makerAmount, rate)
-    const newGoldBalance = getNewGoldBalance(goldBalance, makerToken, makerAmount, takerAmount)
-    const newDollarBalance = getNewDollarBalance(
-      dollarBalance,
-      makerToken,
-      makerAmount,
-      takerAmount
+    const { exchangeRatePair, fee, t, appConnected } = this.props
+
+    const exchangeRate = getRateForMakerToken(
+      exchangeRatePair,
+      this.state.makerToken,
+      CURRENCY_ENUM.DOLLAR
     )
+    const dollarAmount = this.getInputAmountInToken(CURRENCY_ENUM.DOLLAR)
+
+    const buttonText =
+      this.state.makerToken === CURRENCY_ENUM.DOLLAR
+        ? t(`${Namespaces.exchangeFlow9}:buy`)
+        : t(`${Namespaces.exchangeFlow9}:sell`)
 
     return (
       <SafeAreaView style={styles.container}>
-        <DisconnectBanner />
-        <ReviewFrame
-          HeaderComponent={this.renderHeader}
-          confirmButton={{
-            action: this.onPressConfirm,
-            text: t('exchange'),
-            disabled: !appConnected || rate.isZero(),
-          }}
-          modifyButton={{ action: this.onPressEdit, text: t('edit') }}
-        >
-          <ExchangeConfirmationCard
-            makerToken={makerToken}
-            newDollarBalance={newDollarBalance}
-            newGoldBalance={newGoldBalance}
-            makerAmount={makerAmount}
-            takerAmount={takerAmount}
-            exchangeRate={rate}
-            fee={fee}
+        <View style={styles.paddedContainer}>
+          <DisconnectBanner />
+          <ScrollView>
+            <View style={styles.column}>
+              <View style={[styles.rowContainer, styles.amountRow]}>
+                <Text style={[fontStyles.body, styles.exchangeBodyText]}>
+                  {t('amount') + ` (${this.state.inputTokenCode})`}
```

Comment:
As commented above, we should use a single translation key and use interpolation.

Approximate implementation ([using nesting and interpolation](https://www.i18next.com/translation-function/nesting#passing-nesting-to-interpolated), but there are other ways):

```ts
t('exchangeAmount', {tokenName: `$t('global:${ key }')` })
```

Locale json
```json
exchangeAmount: "Amount ({{tokenName}})"
```

Reply:
Fixed